temoa/extensions/framework.py

```
@@ -0,0 +1,231 @@
+            raise RuntimeError(msg)
+
+
+def _table_has_rows(con: Connection, table_name: str) -> bool:
+    cur = con.cursor()
+    table_exists = cur.execute(
+        "SELECT 1 FROM sqlite_master WHERE type='table' AND name = ?", (table_name,)
+    ).fetchone()
+    if not table_exists:
+        return False
+
+    query = f'SELECT 1 FROM main.{table_name} LIMIT 1'
```

Ruff (0.15.21)

[error] Possible SQL injection vector through string-based query construction

(S608)